proposals/repo-to-tar/README.md

```diff
@@ -0,0 +1,106 @@
+
+And also to make it compatible to do both, pushing folders as bundle directly to the registry whilist also obtaining a tar image, we will similarily add a --oci-tar flag : <br>
+`imgpkg push -b registry.example.com/xyz -f some-folder --to-oci-tar local-oci-format.tar`
+
+Oci Tar Pushing : <br>
+
+Now in order to push the tar file to the registry, we can use the already existing copy command with a new flag as follows : <br>
+`imgpkg copy --oci-tar local-oci-format.tar --to-repo registry.example.com/abc`
+
+### Goals and Non-goals and Future-goals
+
+#### Goals
+1. Making OCI Complaint tar and image : <br>
+Create and save the image in oci format locally along with tar creation.
+An example of directory structure formed from complying to [image spec](https://github.com/opencontainers/image-spec/blob/main/image-layout.md) : <br>
+
+```md
+├── blobs
+│   └── sha256
+│       ├── 45e16698e6e721de091b5ecbb811eae1243fb8f2ed5dbd04a9f40cd199c355f9
+│       ├── 4da097607a3cb58407cd8c157271bb5f7462a5872d516a5bc979ec678ac30f7d
+│       └── 5c2dafe3c70c13990190d643c91e9f67b8129b179257674888178868474f6511
+├── index.json
+└── oci-layout
+```
+where one of the files in sha256 is the actual file tar and others 2 are configs and manifests.<br>
+
+2. Add functionality to `imgpkg save` to create and persist a oci tar file mentioned in point 1 which in turn is compatible with `imgpkg copy` command.<br>
+
+3. Add a flag `--to-oci-tar` to `imgpkg push`  to be able to creata a tar while pushing a tar with imgpkg push command.
+ 
+4. Add a flag `--oci-tar` to `imgpkg copy` to be able to copy this oci kind of tar to a registry and differentiate with the `--tar` flag.
+
+#### Future Goals
+
+1. An improvement would be to have only one flag `--tar` for `imgpkg copy` making no changes to flags but making it compatible and being smart enough to understand and differentiate between both kind of tars without any need for specification. <br>
+
+2. Add an inflate option to also contain all refs images in the oci-tar as discussed [here](https://github.com/carvel-dev/imgpkg/issues/55#issuecomment-962209740). This will be useful and true to the name of the command `imgpkg copy` as it will copy all the refs images as well and create a self-sufficient oci-tar. <br>
+
+### Specification / Use Cases
+
+Use Cases : 
+- User wants to build a local oci compatible tar that can be pushed at a later point without being dependent on a registry.
+- While using a CI framework which allows users to share files between stages, user wants to pass this output so that it can be pushed/copied in the next stage rather than having a shared registry.
+- User can do the above with Package and PackageRepository bundles
+
+### Other Approaches Considered
+
+The idea behind the usage of `imgpkg push -b my.registry.io/some-name -f some-folder --to-oci-tar local-oci-format.tar` is to concentrate in a single command all the options to create a new image, that being to the registry or directly to disk.
```

Comment:
My main concern with this is that the `-b` option is _required_ (because we want to add labels if it is bundle), and because of that the first issue that we are trying to solve in this problem still remains ("Dependent on using a registry to create a tar file from a bundle image.")